fix bug : the font selected of mSpinBox object is not effective on runt time.

10km opened the pull request:
Bug descriptsion:
the font selected by `mSpinBox `object is not effective on runt time.

REASON:
`mSpinBox_wndProc `function don't respond message `MSG_FONTCHANGED`,

SOLUTION:

moditiy the `mSpinBox_wndProc `,add respone of `MSG_FONTCHANGED` , set font of child editor.

fixed another issue: 
replace all of hardcode `100`  that  represent for editor id with a new preprocessor `EDITOR_CHILD_ID`

10km committed:
fix bug : the font selected of mSpinBox object is not effective on runt time.

10km committed:
MainWindowThreadCleanup will crash when dialog->hwnd is not main window,
add a judgement statement to avoid the case.

10km commented:
the second commit `e90a5d5` fix another simple issue about dialog box

VincentWei reviewed (approved):
Great! Thanks a lot!

VincentWei reviewed (commented):
What will occur if we do not check whether the font is a system font and do not destroy the old font?

10km committed:
simplify the logic about MSG_FONTCHANGED.
For user custom font,the responsiblity of Destroying Font belong to application.
so mSpinBox_wndProc CAN NOT and have no duty  to CREATE or DESTROY font.

10km commented:
Thanks for your question,I think I make a mistake:
For user custom font,the responsiblity of Destroying Font belong to application.
As part of a facilities ,mSpinBox_wndProc CAN NOT and have no duty  to CREATE or DESTROY font.
So I delete the `is_system_font` function. the logic is more simple and clear.

VincentWei reviewed (approved):
Great!

10km commented:
I have fixed above issue in the commit
[541cf01](https://github.com/VincentWei/mgncs/pull/1/commits/541cf017ebf576cf0a46cc09f02012e8452a80f5) .
need I  create a new pull request?

VincentWei commented:
Ok. Please create a new pull request. I can not merge your change from the commit 541cf01.
Thanks!
